### PR TITLE
[vpj][server][controller][doc] VPJ UX improvement and doc

### PR DESCRIPTION
**Pull request by FelixGV**
Reduced the number of mandatory configs by adding defaults:

- key.field now defaults to "key"
- value.field now defaults to "value"
- venice.ssl.enable now defaults to false (it was partially this way already, but some code still looked for it, so that is now fixed).

Lots of test clean up to take advantage of defaults (though some tests do still override these props to keep these code paths tested).

Other clean ups:
- Eliminated one fairly widespread usage of the deprecated Pair class in TestPushUtils.
- Deleted the AutoClosingKafkaConsumer and related code.
- Added exponential backoff to some busy test assertion.

Initial doc for VPJ.

## How was this PR tested?
Internal CI (yellow)

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 1]
```diff
@@ -602,29 +605,31 @@ PushJobSetting getPushJobSetting() {
...
+    handleLegacyConfig(vanillaProps, LEGACY_AVRO_VALUE_FIELD_PROP, VALUE_FIELD_PROP, "value field");
+    return new VeniceProperties(vanillaProps);
+  }
+
+  private void handleLegacyConfig(
+      Properties vanillaProps,
+      String legacyConfigProp,
+      String newConfigProp,
+      String configDescription) {
+    String legacyConfig = vanillaProps.getProperty(legacyConfigProp);
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:618]: Hmm... I am not sure. I'm afraid it'll break something. Would need to audit the usage first...

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 2]
```diff
@@ -534,9 +540,7 @@ public VenicePushJob(
     initControllerClient(
         props.getString(VENICE_STORE_NAME_PROP),
         getVeniceControllerUrl(props),
-        createSSlFactory(
-            props.getBoolean(ENABLE_SSL, false),
-            props.getString(SSL_FACTORY_CLASS_NAME, DEFAULT_SSL_FACTORY_CLASS_NAME)),
+        createSSlFactory(isSslEnabled(), props.getString(SSL_FACTORY_CLASS_NAME, DEFAULT_SSL_FACTORY_CLASS_NAME)),
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:543]: Done.

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 3]
```diff
@@ -2536,7 +2548,7 @@ protected void setupDefaultJobConf(
       StoreSetting storeSetting,
       VeniceProperties props,
       String id) {
-    setupCommonJobConf(conf, pushJobSchemaInfo, id + ":venice_push_job-" + versionTopicInfo.topic);
+    setupCommonJobConf(conf, id + ":venice_push_job-" + versionTopicInfo.topic);
     conf.set(BATCH_NUM_BYTES_PROP, Integer.toString(pushJobSetting.batchNumBytes));
     conf.set(TOPIC_PROP, versionTopicInfo.topic);
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:2553]: SG, done.

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 4]
```diff
@@ -602,29 +605,31 @@ PushJobSetting getPushJobSetting() {
...
-          && !vanillaProps.getProperty(KEY_FIELD_PROP).equals(vanillaProps.getProperty(LEGACY_AVRO_KEY_FIELD_PROP))) {
-        throw new VeniceException("Duplicate key filed found in config. Both avro.key.field and key.field are set up.");
-      }
-      vanillaProps.setProperty(KEY_FIELD_PROP, vanillaProps.getProperty(LEGACY_AVRO_KEY_FIELD_PROP));
-    }
-    if (vanillaProps.containsKey(LEGACY_AVRO_VALUE_FIELD_PROP)) {
-      if (vanillaProps.containsKey(VALUE_FIELD_PROP) && !vanillaProps.getProperty(VALUE_FIELD_PROP)
-          .equals(vanillaProps.getProperty(LEGACY_AVRO_VALUE_FIELD_PROP))) {
+    handleLegacyConfig(vanillaProps, LEGACY_AVRO_KEY_FIELD_PROP, KEY_FIELD_PROP, "key field");
+    handleLegacyConfig(vanillaProps, LEGACY_AVRO_VALUE_FIELD_PROP, VALUE_FIELD_PROP, "value field");
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:609]: Done.

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 5]
```diff
@@ -1961,6 +1966,11 @@ private void validateValueSchema(
           c -> c.getValueSchemaID(setting.storeName, pushJobSchemaInfo.getValueSchemaString()));
     }
     if (getValueSchemaIdResponse.isError() && !schemaAutoRegisterFromPushJobEnabled) {
+      MultiSchemaResponse response = controllerClient.getAllValueSchema(setting.storeName);
+      LOGGER.info("All currently registered value schemas:");
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:1970]: Done.

File: docs/user_guide/write_api/push_job.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+permalink: /docs/user_guide/write_api/push_job
+---
+# Push Job
+The Push Job takes data from a Hadoop grid and writes it to Venice. 
+
+## Use Cases
+There are two modes the Push Job can run in:
+
+- Full Push (default)
+- Incremental Push
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply docs/user_guide/write_api/push_job.md:14]: I would prefer doing another page dedicated to WC. Then we might want to link to it from here...

File: docs/user_guide/write_api/push_job.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+### Full Push
+When performing a Full Push, the user takes advantage of the fact that Venice's datasets are versioned. The Full Push
+triggers the dynamic creation of a new dataset version, and then loads data into it in the background. The new dataset 
+version is called a "future" version as long as data is still loading, and during that time, no online read traffic will 
+hit it. When the loading is determined to have successfully completed in a given datacenter, the new dataset version
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply docs/user_guide/write_api/push_job.md:20]: Done.

**Review by nisargthakkar — APPROVED** [review]
LGTM

File: docs/user_guide/write_api/push_job.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+The Push Job takes data from a Hadoop grid and writes it to Venice. 
+
+## Use Cases
+There are two modes the Push Job can run in:
+
+- Full Push (default)
+- Incremental Push
+
+### Full Push
+When performing a Full Push, the user takes advantage of the fact that Venice's datasets are versioned. The Full Push
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply docs/user_guide/write_api/push_job.md:17]: I had missed that one in my first pass... will include it in the next commit.

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (FelixGV) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:609]: Instead, I managed to confirm that we don't use the legacy one at all, so I'll just delete this deprecated stuff completely.

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 9]
```diff
@@ -1967,9 +1942,13 @@ private void validateValueSchema(
...
       MultiSchemaResponse response = controllerClient.getAllValueSchema(setting.storeName);
-      LOGGER.info("All currently registered value schemas:");
-      for (MultiSchemaResponse.Schema schema: response.getSchemas()) {
-        LOGGER.info("Schema " + schema.getId() + ": " + schema.getSchemaStr());
+      if (response.isError()) {
+        LOGGER.error("Failed to fetch all value schemas, so they will not be printed.");
+      } else {
+        LOGGER.info("All currently registered value schemas:");
+        for (MultiSchemaResponse.Schema schema: response.getSchemas()) {
+          LOGGER.info("Schema " + schema.getId() + ": " + schema.getSchemaStr());
```
Comment (nisargthakkar) [inline clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:1950]: Let's use interpolation format for logging here

**Review by nisargthakkar — CHANGES_REQUESTED** [review]
Mostly LGTM

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java (hunk 9 above)
Comment (FelixGV) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java]: Done, thanks!

**Review by nisargthakkar — APPROVED** [review]
LGTM

**Pull request merged**